otelcol/command.go

```diff
@@ -73,14 +70,8 @@ func updateSettingsUsingFlags(set *CollectorSettings, flags *flag.FlagSet, enfor
 		set.ConfigProviderSettings.ResolverSettings.DefaultScheme = "env"
 	}
 
-	// Provide a default set of providers and converters if none have been specified.
-	// TODO: Remove this after CollectorSettings.ConfigProvider is removed and instead
-	// do it in the builder.
 	if len(resolverSet.ProviderFactories) == 0 && len(resolverSet.ConverterFactories) == 0 {
```

Comment:
```suggestion
	if len(resolverSet.ProviderFactories) == 0 {
```
I still think we should remove this check and have it handled by the resolver, but if we intend to keep it here, we should probably remove the converter check now that we don't need to look for a "zero-valued" factory config.